src/sagemaker/image_uris.py

```diff
@@ -34,28 +45,48 @@ def retrieve(framework, region, version=None, py_version=None, instance_type=Non
         instance_type (str): The SageMaker instance type. For supported types, see
             https://aws.amazon.com/sagemaker/pricing/instance-types. This is required if
             there are different images for different processor types.
+        accelerator_type (str): Elastic Inference accelerator type. For more, see
+            https://docs.aws.amazon.com/sagemaker/latest/dg/ei.html.
+        image_scope (str): The image type, i.e. what it is used for.
+            Valid values: "training", "inference", "eia". If ``accelerator_type`` is set,
+            ``image_scope`` is ignored.
 
     Returns:
         str: the ECR URI for the corresponding SageMaker Docker image.
 
     Raises:
-        ValueError: If the framework version, Python version, processor type, or region is
-            not supported given the other arguments.
+        ValueError: If the combination of arguments specified is not supported.
     """
-    config = config_for_framework(framework)
+    config = _config_for_framework_and_scope(framework, image_scope, accelerator_type)
     version_config = config["versions"][_version_for_config(version, config, framework)]
 
+    py_version = _validate_py_version_and_set_if_needed(py_version, version_config)
+    version_config = version_config.get(py_version) or version_config
+
     registry = _registry_from_region(region, version_config["registries"])
     hostname = utils._botocore_resolver().construct_endpoint("ecr", region)["hostname"]
 
     repo = version_config["repository"]
-
-    _validate_py_version(py_version, version_config["py_versions"], framework, version)
-    tag = "{}-{}-{}".format(version, _processor(instance_type, config["processors"]), py_version)
+    tag = _format_tag(version, _processor(instance_type, config["processors"]), py_version)
 
     return ECR_URI_TEMPLATE.format(registry=registry, hostname=hostname, repository=repo, tag=tag)
 
 
+def _config_for_framework_and_scope(framework, image_scope, accelerator_type=None):
+    """Loads the JSON config for the given framework and image scope."""
+    config = config_for_framework(framework)
+
+    if accelerator_type:
+        if image_scope not in ("eia", "inference"):
+            logger.info(
```

Comment:
warn?

Reply:
yeah, I went back and forth on this one. I'll go with `warn` then.